### ISSUE TITLE
When does flattening of the name occurs?

### ISSUE THREAD

**Issue by Jym77** [w3c/accname#63]
The description of the algorithm states that the result is a flat string (replacing linebreak with spaces and collapsing adjacent spaces). However, there is no point in the algorithm which specifies that this flattening occurs.

Notably, [Step 2C](https://www.w3.org/TR/accname-1.1/#step2C) returns the value of `aria-label` without flattening (nor trimming); and [Step 2G](https://www.w3.org/TR/accname-1.1/#step2G) return the textual content of text node without flattening nor trimming. In the end (no step number), the result of these steps is appended to the `total accumulated text` without flattening nor trimming, and this `total accumulated text` is used as accessible name without flattening or trimming.

**Comment by accdc** [comment]
I agree this isn't as clear as it should be. I hope to address this when I work on the AccName simplification project.

At present though, this is indicated at the beginning of the section at
https://www.w3.org/TR/accname-1.1/#mapping_additional_nd

"The starting point of the name and description computation is a DOM element. The output is a flat, unstructured string that can be as simple as a single word, or a string of space separated tokens."

So, this needs to be applied just after all of the nodes are processed using the recursive algorithm, on the total result string comprising either the name or description property.

"Flat string 
A string of characters where all carriage returns, newlines, tabs, and form-feeds are replaced with a single space, and multiple spaces are reduced to a single space. The string contains only character data; it does not contain any markup."
https://www.w3.org/TR/accname-1.1/#terminology

**Comment by Jym77** [comment]
Yes, this is not too much of a problem given that flattening is idempotent, so doing it once at the end is enough, and doing it at each step is not useful. 
[Step 2D](https://www.w3.org/TR/accname-1.1/#step2D) does explicitly mention flattening the intermediate result, making this even more confusing 😕

**Comment by MelSumner** [comment]
I _think_ that this issue has been resolved:

> AriaLabel: Otherwise, if the current node has an aria-label [attribute](https://dom.spec.whatwg.org/#concept-attribute) whose value is not undefined, not the empty string, nor, when trimmed of [whitespace](https://infra.spec.whatwg.org/#ascii-whitespace), is not the empty string:
